internal/dag/cache.go

```diff
@@ -46,7 +47,9 @@ type KubernetesCache struct {
 	// Secrets that are referred from the configuration file.
 	ConfiguredSecretRefs []*types.NamespacedName
 
-	ingresses            map[types.NamespacedName]*v1beta1.Ingress
+	ingressesv1beta1     map[types.NamespacedName]*v1beta1.Ingress
+	ingressesv1          map[types.NamespacedName]*networking_v1.Ingress
+	ingressclasses       map[string]*networking_v1.IngressClass
```

Comment:
do we need to cache `IngressClass`?

Reply:
I think it's better to, I don't think there'll be that many objects, and it means we don't have to restart informers if ingressclasses change.

Reply:
It doesn't look like the `IngressClass` cache is actually used anywhere. We can follow up and remove it if it's not needed for any of the v1 functionality.